t/43-scheduling-and-worker-scalability.t

```diff
@@ -154,9 +154,15 @@ subtest 'assign and run jobs' => sub {
         is(scalar @$allocated, $worker_count, 'each worker has a job assigned');
     }
     elsif ($remaining_jobs < 0) {
+        # this branch is uncovered in
+        # https://codecov.io/gh/os-autoinst/openQA/src/master/t/43-scheduling-and-worker-scalability.t
+        # , why should we keep it?
         is(scalar @$allocated, $job_count, 'each job has a worker assigned');
     }
     else {
+        # this branch is uncovered in
+        # https://codecov.io/gh/os-autoinst/openQA/src/master/t/43-scheduling-and-worker-scalability.t
+        # , why should we keep it?
         is(scalar @$allocated, $job_count, 'all jobs assigned and all workers busy');
```

Comment:
This test is configurable. One might configure an equal number of workers and jobs. In this case we expect that the scheduler assigns all jobs and all workers have a job assigned. One might also configure more jobs than workers or more workers than jobs. In these cases the checks are obviously more limited and if there are more jobs than workers we obviously need to schedule multiple times until all jobs are processes.

I really prefer to keep the test configurable because it was helpful in the past to reproduce issues with job assignment locally which we otherwise only saw in production. For that playing around with the parameters was very useful.

You may mark the blocks not executed within the CI as unrecoverable with the notes "only executed when there are more jobs than workers" and "only executed when the number of workers is jobs are equal".

Reply:
ok, done that. It's not pretty but I think it's ok to keep it with my "uncoverable" comments as it's really an exception which we don't have anywhere else in the same way